PSReadLine/Movement.vi.cs

```diff
@@ -196,12 +208,23 @@ public static void GotoColumn(ConsoleKeyInfo? key = null, object arg = null)
         /// </summary>
         public static void GotoFirstNonBlankOfLine(ConsoleKeyInfo? key = null, object arg = null)
         {
-            for (int i = 0; i < _singleton._buffer.Length; i++)
+            if (_singleton.LineIsMultiLine())
             {
-                if (!Char.IsWhiteSpace(_singleton._buffer[i]))
+                var newCurrent = GetFirstNonBlankOfLogicalLinePos(_singleton._current);
+                if (newCurrent != _singleton._current)
                 {
-                    _singleton.MoveCursor(i);
-                    return;
+                    _singleton.MoveCursor(newCurrent);
+                }
+            }
+            else
```

Comment:
Is this branch still needed, or would your new code handle single lines correctly as well?

Reply:
You are right. The multiline case handles the more general single-case as well.